storage/azure/src/main/java/io/aiven/kafka/tieredstorage/storage/azure/MetricCollector.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.storage.azure;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.common.MetricNameTemplate;
+import org.apache.kafka.common.metrics.JmxReporter;
+import org.apache.kafka.common.metrics.KafkaMetricsContext;
+import org.apache.kafka.common.metrics.MetricConfig;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.CumulativeCount;
+import org.apache.kafka.common.metrics.stats.Rate;
+import org.apache.kafka.common.utils.Time;
+
+import io.aiven.kafka.tieredstorage.metrics.SensorProvider;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpPipelineNextSyncPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+public class MetricCollector {
+    private final org.apache.kafka.common.metrics.Metrics metrics;
+
+    private static final String METRIC_GROUP = "azure-blob-metrics";
+
+    final AzureBlobStorageConfig config;
+
+    MetricCollector(final AzureBlobStorageConfig config) {
+        this.config = config;
+
+        final JmxReporter reporter = new JmxReporter();
+
+        metrics = new org.apache.kafka.common.metrics.Metrics(
+            new MetricConfig(), List.of(reporter), Time.SYSTEM,
+            new KafkaMetricsContext("aiven.kafka.server.tieredstorage.azure")
+        );
+    }
+
+    Pattern pathPattern() {
+        final var maybeContainer = config.containerName() != null ? config.containerName() + "/" : "";
+        final var exp = "^/" + config.accountName() + "/" + maybeContainer + "([^/])";
+        return Pattern.compile(exp);
+    }
+
+    MetricsPolicy policy() {
+        return new MetricsPolicy(metrics, pathPattern());
+    }
+
+    static class MetricsPolicy implements HttpPipelinePolicy {
+
+        private static final Pattern UPLOAD_QUERY_PATTERN = Pattern.compile("^comp=(?<comp>\\w+)(&)?.*");
```

Comment:
It's maybe a bit strong assumption that it's in the beginning of the query line. Let's maybe use
```suggestion
        private static final Pattern UPLOAD_QUERY_PATTERN = Pattern.compile("comp=(?<comp>[^&]+)");
```
together with `matcher.find()` instead of `matcher.matches()`?